bridge/tracker/rootchain.go

```diff
@@ -0,0 +1,236 @@
+
+	//	set range of blocks to query
+	queryFilter.SetFromUint64(from.Uint64())
+	queryFilter.SetToUint64(to.Uint64())
+
+	//	set contract addresses and topics
+	for _, contract := range contracts {
+		//	append address
+		queryFilter.Address = append(queryFilter.Address, contract.address)
+
+		//	topics from all contracts must be in Topics[0]
+		queryFilter.Topics = append(queryFilter.Topics, contract.eventIDs())
+	}
+
+	return queryFilter
+}
+
+/* Header types parsed by the client */
+
+//	Ethereum header
+//type ethHeader struct {
+//	Difficulty   string        `json:"difficulty"`
+//	ExtraData    string        `json:"extraData"`
+//	GasLimit     string        `json:"gasLimit"`
+//	GasUsed      string        `json:"gasUsed"`
+//	LogsBloom    types.Bloom   `json:"logsBloom"`
+//	Miner        types.Address `json:"miner"`
+//	Nonce        string        `json:"nonce"`
+//	Number       string        `json:"number"`
+//	ParentHash   types.Hash    `json:"parentHash"`
+//	ReceiptsRoot types.Hash    `json:"receiptsRoot"`
+//	Sha3Uncles   types.Hash    `json:"sha3Uncles"`
+//	StateRoot    types.Hash    `json:"stateRoot"`
+//	Timestamp    string        `json:"timestamp"`
+//	TxRoot       types.Hash    `json:"transactionsRoot"`
+//	MixHash      types.Hash    `json:"mixHash"`
+//	Hash         types.Hash    `json:"hash"`
+//}
+
+//	Polygon-Edge header
+type ethHeader struct {
+	Difficulty   uint64        `json:"difficulty"`
+	ExtraData    string        `json:"extraData"`
+	GasLimit     uint64        `json:"gasLimit"`
+	GasUsed      uint64        `json:"gasUsed"`
+	LogsBloom    types.Bloom   `json:"logsBloom"`
+	Miner        types.Address `json:"miner"`
+	Nonce        string        `json:"nonce"`
+	Number       uint64        `json:"number"`
+	ParentHash   types.Hash    `json:"parentHash"`
+	ReceiptsRoot types.Hash    `json:"receiptsRoot"`
+	Sha3Uncles   types.Hash    `json:"sha3Uncles"`
+	StateRoot    types.Hash    `json:"stateRoot"`
+	Timestamp    uint64        `json:"timestamp"`
+	TxRoot       types.Hash    `json:"transactionsRoot"`
+	MixHash      types.Hash    `json:"mixHash"`
+	Hash         types.Hash    `json:"hash"`
+}
+
+/* 	Rootchain storage (last processed block number) */
+
+//	initRootchainDB creates a new database (or loads existing)
+//	for storing the last processed block's number by the tracker.
+func initRootchainDB(logger hclog.Logger, dbPath string) (storage.Storage, error) {
+	if dbPath == "" {
+		dbPath, _ = os.Getwd()
+		dbPath = filepath.Join(dbPath, "last-processed-block")
+	}
+
+	db, err := leveldb.NewLevelDBStorage(dbPath, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	return db, nil
+}
+
+/*	Rootchain sub object	*/
+
+type cancelSubCallback func() error
+
+//	rootchain subscription object
+type subscription struct {
+	newHeadsCh chan *ethHeader
+	errorCh    chan error
+	cancel     cancelSubCallback
+}
+
+//	newHead returns the subscription's channel for new head events.
+func (s *subscription) newHead() <-chan *ethHeader {
```

Comment:
`newHeadCh` is better?

Reply:
[Resolved](https://github.com/0xPolygon/polygon-edge/pull/386/commits/585d357ff9ff834ac33cc2fd98a060539046f5dc)